graph/src/components/ethereum/types.rs

```diff
@@ -110,48 +82,37 @@ impl EthereumCall {
             block_number: trace.block_number,
             block_hash: trace.block_hash,
             transaction_hash: trace.transaction_hash,
+            transaction_index,
         })
     }
 }
 
-#[derive(Clone, Debug, Deserialize, Serialize)]
+#[derive(Clone, Debug)]
 pub enum EthereumTrigger {
     Block(EthereumBlockTriggerType),
     Call(EthereumCall),
     Log(Log),
 }
 
-#[derive(Clone, Debug, Deserialize, Serialize)]
+#[derive(Clone, Debug)]
 pub enum EthereumBlockTriggerType {
     Every,
     WithCallTo(Address),
 }
 
 impl EthereumTrigger {
-    pub fn transaction_index(
-        &self,
-        transaction_hash_index_lookup: &HashMap<H256, u64>,
-    ) -> Result<Option<u64>, ()> {
+    pub fn transaction_index(&self) -> Option<u64> {
         match self {
-            EthereumTrigger::Log(log) => {
-                match transaction_hash_index_lookup.get(&log.transaction_hash.unwrap()) {
-                    Some(index) => Ok(Some(*index)),
-                    None => Err(()),
-                }
-            }
-            EthereumTrigger::Call(call) => {
-                match transaction_hash_index_lookup.get(&call.transaction_hash.unwrap()) {
-                    Some(index) => Ok(Some(*index)),
-                    None => Err(()),
-                }
-            }
-            EthereumTrigger::Block(_call) => Ok(None),
+            // We only handle logs that are in a block and therefore have a `transaction_index`.
+            EthereumTrigger::Log(log) => Some(log.transaction_index.unwrap().as_u64()),
```

Comment:
Do we know under what conditions logs might _lack_ a transaction index?

Reply:
I believe that only happens for pending transactions.

Reply:
True, that makes sense.